clang/unittests/ASTMatchers/ASTMatchersTraversalTest.cpp

```diff
@@ -6390,6 +6394,7 @@ TEST(HasTemplateArgumentLoc, BindsToSpecializationWithDoubleArgument) {
                               0, hasTypeLoc(loc(asString("double")))))))))));
 }
 
+#if 0
```

Comment:
Same question here :)

Reply:
Ping on this.

Reply:
@erichkeane I just pushed a commit which converts all the `#if`'d out test to use `hasTemplateArgument`/`hasAnyTemplateArgument` instead of `hasTemplateArgumentLoc`/`hasAnyTemplateArgumentLoc`. However, this is a temporary measure.

I want to add support for `ClassTemplateSpecializationDecl`, `VarTemplateSpecializationDecl`, and `FunctionDecl` to `hasTemplateArgumentLoc`/`hasAnyTemplateArgumentLoc` so these tests can be switched back to using the `TemplateArgumentLoc` matchers, but I'm not sure whether it should be included in this PR, or have it's own follow-up PR (I would prefer to include it in this one).